src/core.jl

```diff
@@ -109,6 +118,61 @@ struct Polygon <: Drawable
     vertices::Vector{Point}
 end
 
+"""
+    BoundaryFill <: AbstractPolyFillAlgorithm
+    BoundaryFill(x::Int, y::Int, fill_color::T, boundarycolor::T)
+
+    draw(img, verts, BoundaryFill(x, y, fill_color, boundary_color); connectverts)
+    draw!(img, verts, BoundaryFill(x, y, fill_color, boundary_color); connectverts)
+
+Applies Boundary Filling Algortithm on image inside the vertices provided in order
+
+# Output
+
+Return the boundary filled image inside the vertices(provided in order). 
+
+# Arguments
+
+## `x` && `y`
+
+Boundary fill is a seeded polygon filling algorithm.So we need to provide the seed (x,y) inside the image.
+ // To-do checking if verts and seed are inside the image or note
+ 
+## `fill_color`
+
+The pixels inside the boundary of vertices of polygon are filled with this colorant parameter.
+
+## `boundary_color`
+
+The pixels to define the boundary using verts is filled with this colorant parameter. 
+Without this parameter, the whole picture will be filled with the `fill_color`
+
+```julia
+using ImageDraw
+
+img = zeros(RGB, 7, 7)
+expected = copy(img)
+expected[2:6, 2:6] .= RGB{N0f8}(1)
+verts = [CartesianIndex(2, 2), CartesianIndex(2, 6), CartesianIndex(6, 6), CartesianIndex(6, 2), CartesianIndex(2,2)]
+
+draw(img, verts, BoundaryFill(x=4, y=4, fill_color=RGB(1), boundary_color=RGB(1)); connectverts=true)
+
+```
+"""
+
+struct BoundaryFill{T<:Colorant} <: AbstractPolyFillAlgorithm
+	x::Int
+	y::Int
+	fill_color::T
+    boundary_color::T
+    function BoundaryFill(x::Int, y::Int, fill_color::T, boundary_color::T) where {T<:Colorant}
+        new{T}(x, y, fill_color, boundary_color)
+    end
+end
```

Comment:
I really don't want to mention this, please do not mix the usage of SPACEs and TABs.

Reply:
Okay, I read the Julia's code style suggestions today. I was confused before if we have to use tab or space since both TABs and SPACEs have been used in ImageDraw.jl.
Indentation Level => 4 spaces everywhere

Reply:
Yes, it's a bit annoying when both SPACEs and TABs are around, so whenever these mixture usages are detected, feel free to change them to 4 SPACEs.

Just that don't change unrelated codes (even if you find they're using the wrong indentation), because that would pollute the git history with no meaningful changes. This is exactly why we want to make sure that we format the codes well; we don't have many chances to correct them after committing to the git history.